test/cctest/test_inspector_socket.cc

```diff
@@ -46,32 +44,36 @@ static const char HANDSHAKE_REQ[] = "GET /ws/path HTTP/1.1\r\n"
                                     "Sec-WebSocket-Key: aaa==\r\n"
                                     "Sec-WebSocket-Version: 13\r\n\r\n";
 
-static void dispose_handle(uv_handle_t* handle) {
-  *static_cast<bool*>(handle->data) = true;
-}
-
-static void set_timeout_flag(uv_timer_t* timer) {
-  *(static_cast<bool*>(timer->data)) = true;
-}
+class Timeout {
+public:
+  Timeout(uv_loop_t* loop) : timed_out(false), done_(false) {
+    uv_timer_init(loop, &timer_);
+    uv_timer_start(&timer_, Timeout::setFlag, 5000, 0);
+  }
 
-static uv_timer_t* start_timer(bool* flag) {
-  uv_timer_t* timer = new uv_timer_t();
-  uv_timer_init(&loop, timer);
-  timer->data = flag;
-  uv_timer_start(timer, set_timeout_flag, 5000, 0);
-  return timer;
-}
+  ~Timeout() {
+    uv_timer_stop(&timer_);
+    uv_close(reinterpret_cast<uv_handle_t*>(&timer_), markDone);
+    while (!done_) {
+      uv_run(&loop, UV_RUN_NOWAIT);
+    }
+  }
+  bool timed_out;
+private:
+  static void setFlag(uv_timer_t* timer) {
```

Comment:
Style nit: can you either call this `SetFlag` or `set_flag`?  Ditto for markDone.